databricks_cli/sdk/service.py

```diff
@@ -997,29 +997,43 @@ def deploy(self, pipeline_id=None, id=None, name=None, storage=None, configurati
                 raise TypeError('Expected databricks.Filters() or dict for field filters')
         if allow_duplicate_names is not None:
             _data['allow_duplicate_names'] = allow_duplicate_names
-        return self.client.perform_query('PUT', '/pipelines/{pipeline_id}'.format(pipeline_id=pipeline_id), data=_data, headers=headers)
-    
+        return self.client.perform_query('PUT', '/pipelines/{pipeline_id}'.format(pipeline_id=pipeline_id), data=_data,
+                                         headers=headers)
+
     def delete(self, pipeline_id=None, headers=None):
         _data = {}
-    
-        return self.client.perform_query('DELETE', '/pipelines/{pipeline_id}'.format(pipeline_id=pipeline_id), data=_data, headers=headers)
-    
+
+        return self.client.perform_query('DELETE', '/pipelines/{pipeline_id}'.format(pipeline_id=pipeline_id),
+                                         data=_data, headers=headers)
+
     def get(self, pipeline_id=None, headers=None):
         _data = {}
-    
-        return self.client.perform_query('GET', '/pipelines/{pipeline_id}'.format(pipeline_id=pipeline_id), data=_data, headers=headers)
-    
+
+        return self.client.perform_query('GET', '/pipelines/{pipeline_id}'.format(pipeline_id=pipeline_id), data=_data,
+                                         headers=headers)
+
+    def list(self, pagination=None, headers=None):
+        _data = {}
+        if pagination is not None:
+            if not isinstance(pagination, dict):
+                raise TypeError('Expected databricks.Pagination() or dict for field pagination')
+            _data = {"pagination." + key: value for (key, value) in pagination.items()}
```

Comment:
Note that this is not generated, but it's the best place I could think of to add this line (which is necessary for the request to work).

Reply:
Is there something about the Pagination message or this API that makes this line necessary? As we've been told before, having manual changes here is an anti-pattern. If it's something like proto-messages-don't-work-with-GET-requests, we should change the service generation to do this so it works for all APIs.

Reply:
True, there's nothing specific about this API. But, we do need special handling in the generated code.

Namely, [around this line](https://github.com/databricks/universe/blob/master/api-base/src/main/scala/com/databricks/api/base/PythonProtoClientGenerator.scala#L163) in the code generator needs to change, to path-compress data dicts for GET requests. 

I feel like this is an easier stopgap for now? Doing what I suggested will probably require a much larger discussion.

Reply:
We've needed to work around the SDK generation in other places like https://github.com/databricks/databricks-cli/blob/master/databricks_cli/jobs/api.py#L45. Can we do something similar here?

Reply:
I like Andrew's suggestion better than making manual changes in service.py. Though I'd like to understand the issue a little more, so at the very least we can file a ticket with all the details. But best of all is to be good citizens and fix it ourselves.

Is the problem that the CLI is constructing incorrect requests, or that the server is incorrectly parsing correct requests? Why does your fix actually fix the issue?

Reply:
The former. The code generated by our RPC parser doesn't work for our use-case. Basically, the problem is that for GET requests with nested data dicts , like `{"pagination" : { "max_results": 1 }}` the request the CLI produces is `GET /pipelines?pagination=max_results` which is obviously wrong. 

My fix turns `{"pagination" : { "max_results": 1 }}` into `{"pagination.max_results": 1}, which produces `/pipelines?pagination.max_results=1`, which the server can parse and handle correctly. What we need to do in code generator is this, essentially: recursively flatten all nested dicts for GET requests, while path-compressing the keys.

I think Andrew's idea makes sense here. I'll make a bug for actually fixing this upstream, which should be pretty straightforward.

Reply:
Done.